streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/VariableServiceImpl.java

```diff
@@ -79,4 +110,74 @@ public Variable findByVariableCode(Long teamId, String variableCode) {
     public List<Variable> findByTeamId(Long teamId) {
         return baseMapper.selectByTeamId(teamId);
     }
+
+    /**
+     * Replace placeholders with defined variable codes.
+     * @param teamId
+     * @param paramWithPlaceholders Parameters with placeholders, e.g. "--cluster ${kafka.cluster}"
+     * @return
+     */
+    @Override
+    public String parseVariable(Long teamId, String paramWithPlaceholders) {
```

Comment:
How about rename to `replaceVariable`?

Your comment is clear: `Replace placeholders with defined variable codes.`

And I don't think `paramWithPlaceholders` is a good name. What does `param` mean here? Is methodParam? Or do you think `sql` or `job args` are param?

Reply:
@1996fanrui @wolfboys is this okay？

/**
     * Replace variable with defined variable codes.
     * @param teamId
     * @param mixed Text with placeholders, e.g. "--cluster ${kafka.cluster}"
     * @return
     */
    @Override
    public String replaceVariable(Long teamId, String mixed) {
        if (StringUtils.isEmpty(mixed)) {
            return mixed;
        }
        List<Variable> variables = findByTeamId(teamId);
        if (CollectionUtils.isEmpty(variables)) {
            return mixed;
        }
        Map<String, String> variableMap = variables.stream().collect(Collectors.toMap(Variable::getVariableCode, Variable::getVariableValue));
        String restore = mixed;
        Matcher matcher = placeholderPattern.matcher(restore);
        while (matcher.find()) {
            String placeholder = matcher.group();
            String variableCode = getCodeFromPlaceholder(placeholder);
            String variableVaule = variableMap.get(variableCode);
            if (StringUtils.isNotEmpty(variableVaule)) {
                restore = restore.replace(placeholder, variableVaule);
            }
        }
        return restore;
    }

Reply:
Change variables to uppercase

 
   private final Pattern PLACEHOLDER_PATTERN = Pattern.compile("\\$\\{([A-Za-z])+([A-Za-z0-9._-])+\\}");

    private final String PLACEHOLDER_START = "${";

    private final String PLACEHOLDER_END = "}";

Reply:
> @1996fanrui @wolfboys is this okay？
> 
> ```
> /**
>      * Replace variable with defined variable codes.
>      * @param teamId
>      * @param mixed Text with placeholders, e.g. "--cluster ${kafka.cluster}"
>      * @return
>      */
>     @Override
>     public String replaceVariable(Long teamId, String mixed) {
>         if (StringUtils.isEmpty(mixed)) {
>             return mixed;
>         }
>         List<Variable> variables = findByTeamId(teamId);
>         if (CollectionUtils.isEmpty(variables)) {
>             return mixed;
>         }
>         Map<String, String> variableMap = variables.stream().collect(Collectors.toMap(Variable::getVariableCode, Variable::getVariableValue));
>         String restore = mixed;
>         Matcher matcher = placeholderPattern.matcher(restore);
>         while (matcher.find()) {
>             String placeholder = matcher.group();
>             String variableCode = getCodeFromPlaceholder(placeholder);
>             String variableVaule = variableMap.get(variableCode);
>             if (StringUtils.isNotEmpty(variableVaule)) {
>                 restore = restore.replace(placeholder, variableVaule);
>             }
>         }
>         return restore;
>     }
> ```

LGTM